src/node/stage/approved.rs

```diff
@@ -331,15 +356,37 @@ impl Approved {
                     | Variant::ParsecRequest(..)
                     | Variant::ParsecResponse(..)
                     | Variant::Ping
-                    | Variant::Bounce { .. } => false,
+                    | Variant::Bounce { .. } => Ok(MessageAction::Discard),
                 }
             }
+            Variant::JoinRequest(req) => {
+                if !self.should_handle_join_request(our_id, req) {
+                    return Ok(MessageAction::Bounce);
+                }
+
+                if !self.verify_message(msg)? {
+                    return Ok(MessageAction::Discard);
+                }
 
-            Variant::BootstrapResponse(_) | Variant::NodeApproval(_) | Variant::Ping => false,
+                Ok(MessageAction::Handle)
+            }
+            Variant::BootstrapResponse(_) => Ok(MessageAction::Discard),
+            Variant::BootstrapRequest(_)
+            | Variant::MemberKnowledge(_)
+            | Variant::ParsecRequest(..)
+            | Variant::ParsecResponse(..)
+            | Variant::Bounce { .. } => {
+                if self.verify_message(msg)? {
+                    Ok(MessageAction::Handle)
+                } else {
+                    Ok(MessageAction::Discard)
+                }
+            }
+            Variant::Ping => Ok(MessageAction::Discard),
```

Comment:
seems this entry can now be merged with BootstrapResponse (both just for discarding) ?